basenode/src/main/java/io/coti/basenode/services/CommunicationService.java

```diff
@@ -40,6 +41,35 @@ public void removeSender(String receivingFullAddress, NodeType nodeType) {
         zeroMQSender.disconnectFromNode(receivingFullAddress, nodeType);
     }
 
+    @Override
+    public void senderReconnect(String receivingFullAddress, NodeType nodeType) {
+        if (zeroMQSender.isNewlyConnectedToNode(receivingFullAddress)) {
+            return;
+        }
+        int numberOfReconnectAttempts = 0;
+        do {
+            if (zeroMQSender.isConnectedToNode(receivingFullAddress)) {
+                zeroMQSender.disconnectFromNode(receivingFullAddress, nodeType);
+            }
+            zeroMQSender.connectToNode(receivingFullAddress, nodeType);
+            waitUntilConnected(receivingFullAddress);
+            numberOfReconnectAttempts++;
+        } while (numberOfReconnectAttempts < 3 && !zeroMQSender.isConnectedToNode(receivingFullAddress));
+    }
+
+    private static void waitUntilConnected(String receivingFullAddress) {
+        int numberOfChecks = 0;
+        while (numberOfChecks < 25 && !zeroMQSender.isConnectedToNode(receivingFullAddress)) {
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new CotiRunTimeException(String.format("Error while sleep waiting for sender reconnect, error: %s", e));
```

Comment:
if i remember, this will not throw since the interppt above, plz check in the sleep, put value of 10sec, and do kill to process in background. it shold not reach the throw